js/common_ajax_controller.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * Client code to handle AJAX updates orchestrated by the CommonAjaxController class
+ */
+
+/* global glpi_toast_info */
+/* global glpi_toast_warning */
+/* global glpi_toast_error */
+
+// Isolate functions and run when document is ready
+$(() => {
```

Comment:
Since this is new code, can it be made a module/class from the start?
Hiding everything inside a function like this may make testing the code more complex.
Also, can Jest tests be added?

Reply:
Jest is not super interesting here because we are mainly doing DOM manipulation.
However, E2E tests will be very interesting here (and for the form editor in general).

I'll add a lot of them once your cypress work is merged.